src/components/CommentAuthor.jsx

```diff
@@ -25,34 +25,32 @@ class CommentAuthor extends React.Component {
 
     componentDidMount() {
         this._ismounted = true;
-        this.fetchUserNames();
+        this.fetchCreatedByName();
     }
 
     componentWillUnmount() {
         this._ismounted = false;
     }
 
-    componentDidUpdate() {
-        this.fetchUserNames();
-    }
-
-    fetchUserNames() {
-        if (!this.state.createdByName && this.props.createdBy) {
-            this.fetchUserName(this.props.createdBy, 'createdByName');
+    componentDidUpdate(prevProps) {
+        if (this.props.accessToken
+            && (prevProps.accessToken !== this.props.accessToken
+                || prevProps.createdBy !== this.props.createdBy)) {
+            this.fetchCreatedByName();
         }
     }
 
-    fetchUserName(userId, stateToUpdate) {
-        fetchUserName(this.props.accessToken, userId)
-            .then((responseJson) => {
-                if (!responseJson || !responseJson.profile) {
-                    return;
-                }
-
-                this.safeSetState({
-                    [stateToUpdate]: responseJson.profile.name,
+    fetchCreatedByName() {
+        if (!this.state.createdByName && this.props.createdBy) {
```

Comment:
Same here (as it is copy-pasted code). `fetchCreatedByName` must get the latest data if called regardless of the value in `this.state.createdByName`.

Reply:
Picked up in #123.